src/web-lock.test.ts

```diff
@@ -0,0 +1,75 @@
+import 'navigator.locks';
+import {expect} from '@esm-bundle/chai';
+import {newTab, Tab} from './test-util';
+
+let tab: Tab;
+
+suiteSetup(async () => {
+  tab = await newTab('src/web-lock.test.ts');
+});
+
+suiteTeardown(() => {
+  tab.close();
+});
+
+test('storage', async () => {
+  window.localStorage.setItem('test', 'original');
+  window.sessionStorage.setItem('test', 'original');
+
+  await tab.run('window.sessionStorage.setItem("test", "modified")');
+  expect(window.sessionStorage.getItem('test')).to.equal('original');
+
+  await tab.run('window.localStorage.setItem("test", "modified")');
+  expect(window.localStorage.getItem('test')).to.equal('modified');
+});
+
+export async function lockIfAvailable(name: string): Promise<boolean> {
+  return (
+    (await navigator.locks.request(
+      name,
+      {ifAvailable: true, mode: 'shared'},
+      async lock => lock !== null,
+    )) || false
+  );
+}
+
+test('ifAvailable', async () => {
+  await navigator.locks.request('exlock', {mode: 'shared'}, async () => {
+    expect(await tab.run('await lockIfAvailable("exlock")')).to.equal(true);
```

Comment:
I love how the exported bindings are also loaded into the new tab. Awesome work!

Reply:
Thank you! I went through several drafts where I had different snippets of pure JS in quoted strings, and it was just proving impossible to debug. I'm glad I figured out how to make this work.